### PR TITLE
gh-93121: fix test_mailbox where some test cases were accidentally dropped

### PR DESCRIPTION
**Pull request by AGZain**
Before:

https://github.com/python/cpython/blob/b2694ab46997746eae7e913738623b39f6334473/Lib/test/test_mailbox.py#L34-L35

https://github.com/python/cpython/blob/b2694ab46997746eae7e913738623b39f6334473/Lib/test/test_mailbox.py#L2267-L2290

Since `_sample_headers` is a map with duplicate keys (ex: `Received`), not all keys are tested. 

After:

Changed _sample_headers to a list of tuples so all keys are tested

* Issue: gh-93121

### CODE REVIEW

File: Lib/test/test_mailbox.py [hunk 1]
```diff
@@ -2264,30 +2264,30 @@ def test_nonempty_maildir_both(self):
...
+    ("Received", """by andy.gregorykjohnson.com (Postfix, from userid 1000)
+        id 490CD9DD17; Wed, 13 Jul 2005 17:23:11 -0400 (EDT)"""),
+    ("Date", "Wed, 13 Jul 2005 17:23:11 -0400"),
+    ("From", """"Gregory K. Johnson" <gkj@gregorykjohnson.com>"""),
+    ("To", "gkj@gregorykjohnson.com"),
+    ("Subject", "Sample message"),
+    ("Mime-Version", "1.0"),
+    ("Content-Type", """multipart/mixed; boundary="NMuMz9nt05w80d4+\""""),
+    ("Content-Disposition", "inline"),
+    ("User-Agent", "Mutt/1.5.9i")]
```
Comment (AA-Turner) [inline Lib/test/test_mailbox.py]: Style
```suggestion
    ("User-Agent", "Mutt/1.5.9i"),
]
```

File: Misc/NEWS.d/next/Tests/2022-05-26-03-19-46.gh-issue-93121.tTiPpf.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
+Before:
```
Comment (AA-Turner) [inline Misc/NEWS.d/next/Tests/2022-05-26-03-19-46.gh-issue-93121.tTiPpf.rst]: Please delete this file

**Review by AA-Turner — APPROVED** [review]
The substantive change looks OK, two comments.

A

File: Misc/NEWS.d/next/Tests/2022-05-26-03-19-46.gh-issue-93121.tTiPpf.rst (hunk 2 above)
Comment (AGZain) [reply Misc/NEWS.d/next/Tests/2022-05-26-03-19-46.gh-issue-93121.tTiPpf.rst]: Done.

**Comment by rhettinger** [comment]
FWIW, "map" and "mapping" are two different things.

File: Misc/NEWS.d/next/Tests/2022-05-28-03-52-49.gh-issue-93121.JsYqsI.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Since _sample_headers is a map with duplicate keys (ex: Received), not all keys are tested. Changed _sample_headers to a list of tuples so all keys are tested
```
Comment (iritkatriel) [inline Misc/NEWS.d/next/Tests/2022-05-28-03-52-49.gh-issue-93121.JsYqsI.rst]: There are two copies of the news file, please delete one.

**Review by iritkatriel — APPROVED** [review]

**Pull request merged**